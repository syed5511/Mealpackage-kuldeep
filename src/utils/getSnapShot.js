import React from "react";
import { create, act } from "react-test-renderer";

const matchSnapShot = (json) => {
  expect(json).toMatchSnapshot();
};

const getSnapShot = (Component) => {
  let component;
  return (props = {}, type = "c") => {
    if (type === "c") {
      act(() => {
        component = create(<Component {...props} />);
      });
    } else {
      act(() => {
        component.update(<Component {...props} />);
      });
    }
    matchSnapShot(component.toJSON());
    return component;
  };
};

export default getSnapShot;
