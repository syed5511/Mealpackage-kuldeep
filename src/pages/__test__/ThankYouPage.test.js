import ThankYouPage from "../thankyou";
import getSnapShot from "../../utils/getSnapShot";

test("ThankYou page snapshot", () => {
  const snapshot = getSnapShot(ThankYouPage);
  snapshot({ isMobile: false });
  snapshot({ isMobile: true }, "u");
});
