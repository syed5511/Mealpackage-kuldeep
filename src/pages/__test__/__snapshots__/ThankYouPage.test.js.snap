// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThankYou page snapshot 1`] = `
Array [
  <div
    className="sc-fznyAO bGLbWw"
  >
    <div
      className="sc-AxjAm jhiTqR"
    >
      <div
        className="sc-AxirZ bnSjNE"
      >
        <div
          className="sc-AxiKw kNTLLq stepCompleted completed   whiteBg"
          onClick={[Function]}
        >
          1
        </div>
        <div
          className="sc-AxhCb hUrkqV stepCompleted completed   whiteBg"
          onClick={[Function]}
        >
          Select Plan
        </div>
        <div
          className="sc-AxhUy iRPRoQ stepCompleted completed   whiteBg"
        />
      </div>
      <div
        className="sc-AxirZ bnSjNE"
      >
        <div
          className="sc-AxiKw kNTLLq stepCompleted completed   whiteBg"
          onClick={[Function]}
        >
          2
        </div>
        <div
          className="sc-AxhCb hUrkqV stepCompleted completed   whiteBg"
          onClick={[Function]}
        >
          Sign up
        </div>
        <div
          className="sc-AxhUy iRPRoQ stepCompleted completed   whiteBg"
        />
      </div>
      <div
        className="sc-AxirZ bnSjNE"
      >
        <div
          className="sc-AxiKw kNTLLq stepCompleted completed   whiteBg"
          onClick={[Function]}
        >
          3
        </div>
        <div
          className="sc-AxhCb hUrkqV stepCompleted completed   whiteBg"
          onClick={[Function]}
        >
          Checkout
        </div>
        <div
          className="sc-AxhUy iRPRoQ stepCompleted completed   whiteBg"
        />
      </div>
      <div
        className="sc-AxirZ bnSjNE"
      >
        <div
          className="sc-AxiKw kNTLLq stepCompleted completed  current whiteBg"
          onClick={[Function]}
        >
          4
        </div>
        <div
          className="sc-AxhCb hUrkqV stepCompleted completed  current whiteBg"
          onClick={[Function]}
        >
          Choose your meals
        </div>
      </div>
    </div>
  </div>,
  <div
    className="sc-Axmtr dRsMPp"
  >
    <div
      className="sc-AxmLO edkcDn"
    >
      Your first meal package will expire on 15/10/2020 consume all your 3 meals
    </div>
    <div
      className="sc-fzozJi eJrDzv"
    >
      <img
        alt=""
        className="sc-fzoLsD gUkpKy"
        src="/static/images/confirm.png"
      />
       Thanks for signing up, Ikram Ul Haq!
    </div>
    <button
      className="sc-AxgMl cWmMji sc-fzpans djCCfE package_button btn btn-primary"
      disabled={false}
      type="button"
    >
      Place you're first meal order
       
    </button>
  </div>,
]
`;

exports[`ThankYou page snapshot 2`] = `
Array [
  <div
    className="sc-fznyAO bGLbWw mobile"
  >
    <div
      className="sc-AxjAm jhiTqR"
    >
      <div
        className="sc-AxirZ bnSjNE"
      >
        <div
          className="sc-AxiKw kNTLLq stepCompleted completed mobile  whiteBg"
          onClick={[Function]}
        >
          1
        </div>
        <div
          className="sc-AxhUy iRPRoQ stepCompleted completed mobile  whiteBg"
        />
      </div>
      <div
        className="sc-AxirZ bnSjNE"
      >
        <div
          className="sc-AxiKw kNTLLq stepCompleted completed mobile  whiteBg"
          onClick={[Function]}
        >
          2
        </div>
        <div
          className="sc-AxhUy iRPRoQ stepCompleted completed mobile  whiteBg"
        />
      </div>
      <div
        className="sc-AxirZ bnSjNE"
      >
        <div
          className="sc-AxiKw kNTLLq stepCompleted completed mobile  whiteBg"
          onClick={[Function]}
        >
          3
        </div>
        <div
          className="sc-AxhUy iRPRoQ stepCompleted completed mobile  whiteBg"
        />
      </div>
      <div
        className="sc-AxirZ bnSjNE"
      >
        <div
          className="sc-AxiKw kNTLLq stepCompleted completed mobile current whiteBg"
          onClick={[Function]}
        >
          4
        </div>
        <div
          className="sc-AxhCb hUrkqV stepCompleted completed mobile current whiteBg"
          onClick={[Function]}
        >
          Choose your meals
        </div>
      </div>
    </div>
  </div>,
  <div
    className="sc-Axmtr dRsMPp"
  >
    <div
      className="sc-AxmLO edkcDn"
    >
      Your first meal package will expire on 15/10/2020 consume all your 3 meals
    </div>
    <div
      className="sc-fzozJi eJrDzv"
    >
      <img
        alt=""
        className="sc-fzoLsD gUkpKy"
        src="/static/images/confirm.png"
      />
       Thanks for signing up, Ikram Ul Haq!
    </div>
    <button
      className="sc-AxgMl cWmMji sc-fzpans djCCfE package_button btn btn-primary"
      disabled={false}
      type="button"
    >
      Place you're first meal order
       
    </button>
  </div>,
]
`;
